### PR TITLE
Move remaining os.paths over to Path

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

- Move the remaining `os.path` calls over to `pathlib.Path`
- add some rudimentary tests for the `StaticNarrativeCreator` class.

# Jira Ticket / Issue #

Related Jira ticket: https://kbase-jira.atlassian.net/browse/DATAUP-X
- [ ] Added the Jira Ticket to the title of the PR (e.g. `DATAUP-69 Adds a PR template`)

# Testing Instructions
* Details for how to test the PR:
- [x] Tests pass locally and in GitHub Actions

# Dev Checklist:

- [x] My code follows the guidelines at https://sites.google.com/lbl.gov/trussresources/home?authuser=0
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] (Python) I have run Ruff on changed Python code
- [ ] If appropriate, I have recompiled the app and added the updated `StaticNarrativeImpl.py`, `StaticNarrativeServer.py`, and `compile_report.json` to this PR.

# Updating Version and Release Notes (if applicable)

- [ ] [Version has been bumped](https://semver.org/) for each release
- [ ] [Release notes](/RELEASE_NOTES.md) have been updated for each release (and during the merge of feature branches)

### CODE REVIEW

File: lib/StaticNarrative/creator.py [hunk 1]
```diff
@@ -49,60 +49,6 @@ def __init__(self: "StaticNarrativeCreator", config: dict[str, Any], token: str)
         ch.setFormatter(formatter)
         self.logger.addHandler(ch)
 
-    def get_narrative_id_from_workspace(self: "StaticNarrativeCreator", ws_ref: str | int) -> str:
```
Comment (ialarmedalien) [inline lib/StaticNarrative/creator.py:52]: moved down to bottom of file

File: test/test_creator.py [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        lambda _x, _y: [],
+    )
+
+    with pytest.raises(
+        ValueError, match="Workspace 12345 did not contain a KBaseNarrative.Narrative object."
+    ):
+        creator.get_narrative_id_from_workspace(12345)
+
+
+def test_create_local_static_narrative(
```
Comment (ialarmedalien) [inline test/test_creator.py:99]: this test is 100% 💩

File: lib/StaticNarrative/narrative/narrative_util.py [hunk 3]
```diff
@@ -17,26 +17,19 @@
 
 
 def read_narrative(ws_client: Workspace, ref: NarrativeRef) -> dict[str, Any]:
-    """Fetches a Narrative and its object info from the Workspace.
-
```
Comment (ialarmedalien) [inline lib/StaticNarrative/narrative/narrative_util.py:21]: None of the following text is relevant

File: test/test_creator.py (hunk 2 above)
Comment (briehl) [reply test/test_creator.py:99]: You can be sure the mocks work!

Comment (briehl) [reply test/test_creator.py:99]: And that permissions are being checked!

**Review by briehl — APPROVED** [review]

**Pull request merged**